src/normalizers/keywordProtector.js

```diff
@@ -0,0 +1,34 @@
+import { regexMatchingDecoderMaker } from './regex.js';
+
+export function keywordProtector (keywordMap) {
+    const encoder = regexMatchingEncoderMaker(
+        'keywordProtectorDecoder',
+        /(?<protector>protector:\w+)/g,
+        Object.fromEntries(Object.keys(keywordMap).map(kw => [ `protector:${kw}`, keywordMap[kw] ]))
+    );
+    const decoder = regexMatchingDecoderMaker(
+        'keywordProtectorDecoder',
+        new RegExp(`(?<protector>${Object.keys(keywordMap).join("|")||"NO_PROTECTED_STRINGS"})`, 'g'),
+        (groups) => ({ t: 'x', v: `protector:${groups.protector}` })
+    );
+    return [ decoder, encoder ]
+}
+
+//Checks for values in the map based on the flags
+function findFlagValue (charMap, flags) {
+    const v = !charMap || typeof charMap === 'string' || Object.keys(charMap).length === 0 ? charMap : Object.values(flags).find((v) => charMap[v]);
+    return !charMap ? charMap : !charMap[v] || typeof charMap[v] === 'string' ? charMap[v] : findFlagValue (charMap[v], flags);
+}
+ 
+// Generic pluggable encoder that can process values based on flags
+function regexMatchingEncoderMaker(name, regex, charMap) {
```

Comment:
this is the clone of another function. can you refactor the original so it takes a generic callback function if charMap doesn't contain the entry? then you can use it from here passing an empty charMap and a function that looks into your map.

Reply:
I refactored the original, but the logic is if cb exists, use the callback, otherwise the default is charMap[charToReplace].  The reason is that if charMap[charToReplace] exists but isn't a string, it should use the callback.